server/src/lib/db.ts

```diff
@@ -379,17 +382,28 @@ export const createSession = async (userInfo: any) => {
     const sessionId = uuid();
     await sessions.insertOne({
         sessionId,
+        clientCode: clientCode || null,
         userRef: user._id,
     });
 
     return sessionId;
 };
 
+export const updateSession = async (session: any) => {
+    const sessions = db.collection('sessions');
+    return sessions.replaceOne({ _id: session._id }, session, { upsert: true });
```

Comment:
Might want to check whether `session._id` exists? Otherwise `undefined` will be used as key